### PR TITLE
Implemented a moblimiter

### LINKED ISSUE #84: Add a "moblimiter" type feature.
TFM had something like this. Add a feature that can be configured from a config file and/or from in-game that allows admins to toggle the maximum amount of mobs or entities that can be spawned. 

This is very useful for preventing situations where players lag the server by spawning 5000 entities or mobs. It is currently extremely easy to lag the server with massive amounts of entities that can be spawned.

**Example showing why this should be added:**
image
image

### PR DESCRIPTION
**Pull request by notsceptor**
_(This PR resolves #84)_

The mob limiter stores the ceiling value and the max limit in config and can be modified using a command.
_!! It's important to note that this implementation accounts for only living entities !!_

It works on a per chunk basis, here are the default values:
`mob_limit_enabled`: true
`max_mobs_per_chunk`: 50
`mob_limit_ceiling`: 500

The usage for the command `/moblimit` is as follows:
 - `/moblimit` - Shows information about the limiter, whether it's enabled/disabled, the entities per chunk count, the max limit, the chunk coords
 - `/moblimit on` - Toggles the limiter on
 - `/moblimit off` - Toggles the limiter off
 - `/moblimit setmax <limit>` - Sets the max amount of mobs per chunk

### CODE REVIEW

File: server/src/main/java/dev/plex/command/impl/MobLimitCMD.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    {
+        if (silentCheckPermission(sender, this.getPermission()))
+        {
+            if (args.length == 1)
+            {
+                return Arrays.asList("on", "off", "setmax");
+            }
+            if (args.length == 2 && args[0].equals("setmax"))
+            {
+                return Collections.singletonList("<limit>");
```
Comment (Telesphoreo) [inline server/src/main/java/dev/plex/command/impl/MobLimitCMD.java:92]: should not return the usage, just return nothing

File: server/src/main/java/dev/plex/util/PlexUtils.java [hunk 2]
```diff
@@ -297,6 +299,13 @@ public static List<String> getPlayerNameList()
         return Bukkit.getOnlinePlayers().stream().map(Player::getName).collect(Collectors.toList());
     }
 
+    public static boolean isEntityLimitReached(Chunk chunk, int limit)
```
Comment (Telesphoreo) [inline server/src/main/java/dev/plex/util/PlexUtils.java:302]: does this have to go in PlexUtils?

File: server/src/main/java/dev/plex/command/impl/MobLimitCMD.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+                        sender.sendMessage(PlexUtils.messageComponent("mobLimitCeiling"));
+                    }
+
+                    plugin.config.set("entity_limit.max_mobs_per_chunk", newLimit);
+                    plugin.config.save();
+                    return PlexUtils.messageComponent("mobLimitSet", newLimit);
+                }
+                catch (NumberFormatException e)
+                {
+                    return PlexUtils.messageComponent("invalidMobLimit");
```
Comment (Telesphoreo) [inline server/src/main/java/dev/plex/command/impl/MobLimitCMD.java:74]: is there anything in messages.yml that already checks for invalid numbers?

**Review by Telesphoreo — CHANGES_REQUESTED** [review]

File: server/src/main/java/dev/plex/command/impl/MobLimitCMD.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        if (args.length == 0)
+        {
+            Chunk chunk = playerSender != null ? playerSender.getLocation().getChunk() : Bukkit.getWorlds().get(0).getChunkAt(0, 0);
+
+            int currentLimit = plugin.config.getInt("entity_limit.max_mobs_per_chunk");
+            int currentMobCount = (int) Arrays.stream(chunk.getEntities())
+                    .filter(entity -> entity instanceof LivingEntity && !(entity instanceof Player))
+                    .count();
+
+            String status = plugin.config.getBoolean("entity_limit.mob_limit_enabled") ? "<green>Enabled" : "<red>Disabled";
```
[the remark this answers is not in the document]
Comment (notsceptor) [reply server/src/main/java/dev/plex/command/impl/MobLimitCMD.java:37]: there is but it's unformatted and wouldn't work properly in the way I want it to work

File: server/src/main/java/dev/plex/command/impl/MobLimitCMD.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+            String chunkCoords = String.format("<gray>(<em><white>Chunk<gray>: <reset>%d, %d<gray>)", chunk.getX(), chunk.getZ());
```
[the remark this answers is not in the document]
Comment (notsceptor) [reply server/src/main/java/dev/plex/command/impl/MobLimitCMD.java]: added

File: server/src/main/java/dev/plex/command/impl/MobLimitCMD.java (hunk 1 above)
Comment (notsceptor) [reply server/src/main/java/dev/plex/command/impl/MobLimitCMD.java]: changed

File: server/src/main/java/dev/plex/util/PlexUtils.java (hunk 2 above)
Comment (notsceptor) [reply server/src/main/java/dev/plex/util/PlexUtils.java]: moved into the listener

File: server/src/main/java/dev/plex/command/impl/MobLimitCMD.java (hunk 3 above)
Comment (notsceptor) [reply server/src/main/java/dev/plex/command/impl/MobLimitCMD.java]: used a preset parser and removed the invalidMobLimit

**Review by Telesphoreo — APPROVED** [review]

**Pull request merged**